flake.nix

```diff
@@ -76,14 +78,46 @@
           # ];
         };
 
-        packages = flake-utils.lib.flattenTree {
-          docker = import ./nix/docker.nix {
+        packages = flake-utils.lib.flattenTree rec {
+          bot-image = import ./nix/docker.nix {
             inherit pkgs;
             cofree-bot = hsPkgs.cofree-bot;
           };
+
+          repls = import ./nix/repl-containers.nix {
+            inherit pkgs;
+          };
+
           cofree-bot = hsPkgs.cofree-bot;
+
+          deploy-bot =
+            pkgs.writeScript "deploy-bot"
+              ''
+                #!${pkgs.bash}/bin/bash
+                set -euxo pipefail
+
+                image=$(${pkgs.docker}/bin/docker load -i ${bot-image} | sed -n 's#^Loaded image: \([a-zA-Z0-9\.\/\-\:]*\)#\1#p')
+                ${pkgs.docker}/bin/docker push $image
+
+                for x in ${repls}/*; do
+                  image=$(${pkgs.docker}/bin/docker load -i $x | sed -n 's#^Loaded image: \([a-zA-Z0-9\.\/\-\:]*\)#\1#p')
+                  ${pkgs.docker}/bin/docker push $image
+                done
+              '';
+
+          deploy-repls =
+            pkgs.writeScript "deploy-bot"
+              ''
+                #!${pkgs.bash}/bin/bash
+                set -euxo pipefail
+                for x in ${repls}/*; do
+                  ${pkgs.docker}/bin/docker load -i $x
+                done
+              '';
```

Comment:
@JonathanLorimer We are building these scripts for building deploy scripts for the bot and the repl images. We tried to make these apps rather then packages with `pkgs.runCommand` but werent able to run `docker load`. Do you know of any tricks to perform side effects in a flake output or is that off the table?

Reply:
Also we need to get these images onto the server and loaded into docker. I don't want to put them on the github  docker hub because they are going to take up too much space as we add more images. Can we modify our github deploy action to sync them onto the server directly?